gunrock/applications/sssp/sssp_implementation.hxx

```diff
@@ -178,20 +144,62 @@ struct sssp_enactor_t : enactor_t<problem_t> {
     operators::advance::execute<operators::advance_type_t::vertex_to_vertex,
                                 operators::advance_direction_t::forward,
                                 operators::load_balance_t::merge_path>(
-        G, enactor_type::get_enactor(), shortest_path, context);
+        G, E, shortest_path, context);
 
     // Execute filter operator on the provided lambda
     operators::filter::execute<operators::filter_type_t::uniquify>(
-        G, enactor_type::get_enactor(), remove_completed_paths, context);
+        G, E, remove_completed_paths, context);
   }
 
-  sssp_enactor_t(problem_t* _problem,
-                 std::shared_ptr<cuda::multi_context_t> _context)
-      : enactor_type(_problem, _context) {}
-
-  sssp_enactor_t(const sssp_enactor_t& rhs) = delete;            // Boilerplate? Can remove?
-  sssp_enactor_t& operator=(const sssp_enactor_t& rhs) = delete; // Boilerplate? Can remove?
-};  // struct sssp_enactor_t
+};  // struct enactor_t
+
+
+// !! Helper -- This should go somewhere else -- @neoblizz, where?
+auto get_default_context() {
+  std::vector<cuda::device_id_t> devices;
+  devices.push_back(0);
+
+  return std::shared_ptr<cuda::multi_context_t>(
+      new cuda::multi_context_t(devices));
+}
+
+template <
+  typename graph_vector_t,
+  typename meta_vector_t,
+  typename graph_t = typename graph_vector_t::value_type,
+  typename meta_t  = typename meta_vector_t::value_type>
+float run(
+  graph_vector_t& G,
+  meta_vector_t& meta,
+  typename meta_t::vertex_type& single_source, // Parameter
+  typename meta_t::weight_type* distances,     // Output
+  typename meta_t::vertex_type* predecessors   // Output
+) {
+  
+  // <user-defined>
+  param_t<meta_t>  param(single_source);
+  result_t<meta_t> result(distances, predecessors);
+  // </user-defined>
+  // <boiler-plate>
+  auto multi_context = get_default_context();
+
+  using problem_type = problem_t<graph_t, meta_t, param_t<meta_t>, result_t<meta_t>>;
+  using enactor_type = enactor_t<problem_type>;
+  
+  problem_type problem(
+      G.data().get(),    // input graph (GPU)
+      meta.data(),       // metadata    (CPU)
+      &param,            // input parameters
+      &result,           // output results
+      multi_context      // input context
+  );
+  problem.init();
+  problem.reset();
```

Comment:
Another reason why I don't prefer this is if we were to hide multi-GPU stuff, it will be nice to do that transparently for each context in the construction. But I can see ways to use this in a somewhat similar way.